src/Ztm.WebApi/TransactionConfirmationWatcher.cs

```diff
@@ -290,27 +286,24 @@ async Task<bool> StopTimer(Rule rule)
 
         async Task ConfirmAsync(TransactionWatch<Rule> watch, CancellationToken cancellationToken)
         {
-            var rule = await this.ruleRepository.GetAsync(watch.Context.Id, cancellationToken);
-
-            await this.ruleRepository.CompleteAsync(watch.Context.Id, cancellationToken);
-            await ExecuteCallbackAsync(rule.Callback, rule.Success, CancellationToken.None);
+            await this.ruleRepository.UpdateStatusAsync(watch.Context.Id, TransactionConfirmationWatchingRuleStatus.Success, cancellationToken);
+            await ExecuteCallbackAsync(watch.Context.Callback, watch.Context.Success, CancellationToken.None);
         }
 
         async Task ExecuteCallbackAsync(Callback callback, TransactionConfirmationCallbackResult payload, CancellationToken cancellationToken)
         {
-            var id = await this.callbackRepository.AddHistoryAsync(callback.Id, payload, cancellationToken);
+            await this.callbackRepository.AddHistoryAsync(callback.Id, payload, cancellationToken);
 
             try
             {
                 await this.callbackExecuter.Execute(callback.Id, callback.Url, payload);
             }
-            catch (HttpRequestException ex) // lgtm[cs/empty-catch-block]
+            catch (Exception ex) // lgtm [cs/catch-of-all-exceptions]
```

Comment:
We don't need to catch here if we catch at the root level.

Reply:
After discussion we would catch here because we should not deny service because of http request fail.